fleetspeak/src/e2etesting/lib/setup_components.go

```diff
@@ -272,30 +276,66 @@ func modifyFleetspeakClientConfig(configDir string, httpsListenPort int, newLinu
 	return nil
 }
 
-func (cc *ComponentsInfo) start(configDir string, msPort int, numServers, numClients int) error {
-	firstAdminPort := 6060
+// BuildConfigurations builds Fleetspeak configuration files for provided servers and
+// number of clients that are supposed to be started on different machines
+func BuildConfigurations(configDir string, serverHosts []string, numClients int, mysqlCredentials MysqlCredentials) error {
+	err := buildBaseConfiguration(configDir, mysqlCredentials, serverHosts)
+	if err != nil {
+		return fmt.Errorf("Failed to build base configuration: %v", err)
+	}
+
+	httpsListenPort := 6060
+	adminPort := httpsListenPort + 1
+	frontendPort := httpsListenPort + 2
+
+	// Build server configs
+	for i := 0; i < len(serverHosts); i++ {
+		serverConfigPath := path.Join(configDir, fmt.Sprintf("server%v.config", i))
+		serverServicesConfigPath := path.Join(configDir, fmt.Sprintf("server%v.services.config", i))
+		err := modifyFleetspeakServerConfig(configDir, serverHosts[i], frontendPort, httpsListenPort, adminPort, serverConfigPath, serverServicesConfigPath)
+		if err != nil {
+			return fmt.Errorf("Failed to build FS server configurations: %v", err)
+		}
+	}
+
+	// Build client configs
+	for i := 0; i < numClients; i++ {
+		serverIdx := i % len(serverHosts)
```

Comment:
I guess this is ok to start with. But it'd be far more realistic (and make this a more useful example for people aiming to set up a real system) to turn up a load balancer instead of assigning clients to servers manually.

Reply:
Yes, this manual assignment is a quick temporary solution. A load balancer will be added in the next phase of this project.